src/main/java/de/hysky/skyblocker/skyblock/events/EventNotifications.java

```diff
@@ -0,0 +1,171 @@
+                                                                        );
+                                                                    }
+                                                                    return 0;
+                                                                }
+                                                        )
+                                                )
+                                )
+                        )
+                )
+        ));
+    }
+
+    private static final Map<String, LinkedList<SkyblockEvent>> events = new Object2ObjectOpenHashMap<>();
+
+    public static Map<String, LinkedList<SkyblockEvent>> getEvents() {
+        return events;
+    }
+
+    public static void refreshEvents() {
+        CompletableFuture.supplyAsync(() -> {
+            try {
+                JsonArray jsonElements = SkyblockerMod.GSON.fromJson(Http.sendGetRequest("https://hysky.de/api/calendar"), JsonArray.class);
+                return jsonElements.asList().stream().map(JsonElement::getAsJsonObject).toList();
+            } catch (Exception e) {
+                LOGGER.error("[Skyblocker] Failed to download events list", e);
+            }
+            return List.<JsonObject>of();
+        }).thenAccept(eventsList -> {
+            events.clear();
+            for (JsonObject object : eventsList) {
+                if (object.get("timestamp").getAsLong() + object.get("duration").getAsInt() < currentTime) continue;
+                SkyblockEvent skyblockEvent = SkyblockEvent.of(object);
+                events.computeIfAbsent(object.get("event").getAsString(), s -> new LinkedList<>()).add(skyblockEvent);
+            }
+
+            for (Map.Entry<String, LinkedList<SkyblockEvent>> entry : events.entrySet()) {
+                entry.getValue().sort(Comparator.comparingLong(SkyblockEvent::start)); // Sort just in case it's not in order for some reason in API
+                //LOGGER.info("Next {} is at {}", entry.getKey(), entry.getValue().peekFirst());
+            }
+
+            for (String s : events.keySet()) {
+                SkyblockerConfigManager.get().eventNotifications.eventsReminderTimes.computeIfAbsent(s, s1 -> DEFAULT_REMINDERS);
+            }
+        }).exceptionally(EventNotifications::itBorked);
+    }
+
+    private static Void itBorked(Throwable throwable) {
+        LOGGER.error("[Skyblocker] Event loading borked, sowwy :(", throwable);
+        return null;
+    }
+
+
+    private static void timeUpdate() {
+
+        long newTime = System.currentTimeMillis() / 1000;
+        for (Map.Entry<String, LinkedList<SkyblockEvent>> entry : events.entrySet()) {
+            LinkedList<SkyblockEvent> nextEvents = entry.getValue();
+            SkyblockEvent skyblockEvent = nextEvents.peekFirst();
+            if (skyblockEvent == null) continue;
+            if (newTime > skyblockEvent.start() + skyblockEvent.duration()) {
```

Comment:
This might need to be a while loop? Also need to add a label to the continue below if this is changed to a while loop. But if we assume there's only one instance of an event active at a time, then this is fine.

Reply:
if there is 2 events at the same time at some point I will commit acts or arson against hypixel's HQ